doc/signal.md

```diff
@@ -0,0 +1,24 @@
+# Signals
+
+Seastar provides an interface to handle signals natively and safely as asynchronous tasks.
+
+It provides a dedicated header called [seastar/core/signal.hh](../include/seastar/core/signal.hh).
+
+## Usage
+
+You need to construct a `signal_handler` instance that will register the provided signal handler for the specified signal based on the configuration params.
+
+At the moment, the signal handler will remain past object destruction.
```

Comment:
That's extremely confusing. I would expect that desctruction of an object unregisters the handler

Reply:
Ill add unregister logic to this PR